source/containers.tex

```diff
@@ -16086,6 +16093,1215 @@
+    template<class K> pair<iterator, bool> insert(K&& x);
+    iterator insert(const_iterator position, const value_type& x)
+      { return emplace_hint(position, x); }
+    iterator insert(const_iterator position, value_type&& x)
+      { return emplace_hint(position, std::move(x)); }
+    template<class K> iterator insert(const_iterator hint, K&& x);
+
+    template<class InputIterator>
+      void insert(InputIterator first, InputIterator last);
+    template<class InputIterator>
+      void insert(sorted_unique_t, InputIterator first, InputIterator last);
+    template<@\placeholder{container-compatible-range}@<value_type> R>
+      void insert_range(R&& rg);
+
+    void insert(initializer_list<key_type> il)
+      { insert(il.begin(), il.end()); }
+    void insert(sorted_unique_t s, initializer_list<key_type> il)
+      { insert(s, il.begin(), il.end()); }
+
+    container_type extract() &&;
+    void replace(container_type&&);
+
+    iterator erase(iterator position);
+    iterator erase(const_iterator position);
+    size_type erase(const key_type& x);
+    template<class K> size_type erase(K&& x);
+    iterator erase(const_iterator first, const_iterator last);
+
+    void swap(flat_set& y) noexcept;
+    void clear() noexcept;
+
+    // observers
+    key_compare key_comp() const;
+    value_compare value_comp() const;
+
+    // set operations
+    iterator find(const key_type& x);
+    const_iterator find(const key_type& x) const;
+    template<class K> iterator find(const K& x);
+    template<class K> const_iterator find(const K& x) const;
+
+    size_type count(const key_type& x) const;
+    template<class K> size_type count(const K& x) const;
+
+    bool contains(const key_type& x) const;
+    template<class K> bool contains(const K& x) const;
+
+    iterator lower_bound(const key_type& x);
+    const_iterator lower_bound(const key_type& x) const;
+    template<class K> iterator lower_bound(const K& x);
+    template<class K> const_iterator lower_bound(const K& x) const;
+
+    iterator upper_bound(const key_type& x);
+    const_iterator upper_bound(const key_type& x) const;
+    template<class K> iterator upper_bound(const K& x);
+    template<class K> const_iterator upper_bound(const K& x) const;
+
+    pair<iterator, iterator> equal_range(const key_type& x);
+    pair<const_iterator, const_iterator> equal_range(const key_type& x) const;
+    template<class K>
+      pair<iterator, iterator> equal_range(const K& x);
+    template<class K>
+      pair<const_iterator, const_iterator> equal_range(const K& x) const;
+
+    friend bool operator==(const flat_set& x, const flat_set& y);
+
+    friend @\exposconcept{synth-three-way-result}@<value_type>
+    operator<=>(const flat_set& x, const flat_set& y);
+
+    friend void swap(flat_set& x, flat_set& y) noexcept { x.swap(y); }
+
+  private:
+    container_type @\exposid{c}@;           // \expos
+    key_compare @\exposid{compare}@;        // \expos
```

Comment:
```suggestion
    container_type c;           // \expos
    key_compare compare;        // \expos
```

Reply:
These are exposition-only names, I think. Yes, we're locally inconsistent.

Reply:
`flat_map` and `flat_multimap`'s do not use `\exposid` (nor do any container in [container], AFAIK). And my suggestions to drop `\exposid` on them are far less than what you'd have to add to account for the many that were left without `\exposid`.

Reply:
> And my suggestions to drop `\exposid` on them are far less than what you'd have to add to account for the many that were left without `\exposid`.

And by this, I mean the many in just this PR.

Reply:
I've noted the issue, but given the current inconsistencies, don't want to move any more things around until we take a look at the whole text. Leave it as is for now, and we'll clean it up later.